lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -993,6 +995,18 @@ data ApiPostRandomAddressData = ApiPostRandomAddressData
     } deriving (Eq, Generic, Show)
       deriving anyclass NFData
 
+data ApiWalletMigration (n :: NetworkDiscriminant) = ApiWalletMigration
+    { selections :: !(NonEmpty (ApiCoinSelection n))
+    , cost :: !(Quantity "lovelace" Natural)
+    , balanceIncluded :: !ApiWalletMigrationBalance
+    , balanceLeftover :: !ApiWalletMigrationBalance
+    } deriving (Eq, Generic, Show)
+
+data ApiWalletMigrationBalance = ApiWalletMigrationBalance
+    { ada :: !(Quantity "lovelace" Natural)
+    , assets :: !(ApiT W.TokenMap)
+    } deriving (Eq, Generic, Show)
+
 data ApiWalletMigrationPostData (n :: NetworkDiscriminant) (s :: Symbol) =
     ApiWalletMigrationPostData
     { passphrase :: !(ApiT (Passphrase s))
```

Comment:
passphrase wouldn't be needed here, but in the next step (i.e. confirming migration) if I'm not mistaken?
![Screenshot from 2021-03-31 13-21-24](https://user-images.githubusercontent.com/42900201/113136771-0973e400-9224-11eb-959a-ff1dd0ccf7a9.png)

Reply:
Also, is POST the correct verb now when the selection is only returned and not submitted?

I guess that depends on whether we mark the tx as pending immediately or not, which was a point of uncertainty in https://jira.iohk.io/browse/ADP-342

Reply:
> I guess that depends on whether we mark the tx as pending immediately or not.

The intention is for the user to:

1. Call the `migrateWallet` endpoint, which supplies a list of selections (= unsigned transactions).
2. Inspect the results, check that the total fee and the list of selections are acceptable.
3. Call another endpoint (yet-to-be-added), which accepts a list of selections. In the case of HW wallets, the user would sign them first.

The yet-to-be-added endpoint would check that all the supplied selections are valid, by checking that all the UTxO entries still exist, before submitting them as transactions to the network.

So I think there's no need for the `migrateWallet` endpoint to mark transactions as pending here.

If the user calls the `migrateWallet` endpoint and then makes another transaction that spends some of the UTxO entries in the meantime, they can easily get themselves unstuck by calling `migrateWallet` again to generate a new set of selections that don't include those UTxO entries.

Reply:
> Also, is POST the correct verb now when the selection is only returned and not submitted?

I suspect POST is still the correct verb, as the user still has to post a list of target addresses so that we can generate real coin selections with `Address, TokenBundle` pairs as the outputs.

Reply:
> passphrase wouldn't be needed here, but in the next step (i.e. confirming migration) if I'm not mistaken?

Ah yes, well-spotted!

You're right: only the final step (calling the yet-to-be-added submission endpoint) should require a passphrase.

Reply:
> passphrase wouldn't be needed here, but in the next step (i.e. confirming migration) if I'm not mistaken?

Fixed in ed63066f2360baaaef90fb15337e636207885e74.

Reply:
@piotr-iohk @Anviking 

This has changed: we now keep the existing API endpoints (without changing them), but just modify the `ApiWalletMigrationInfo` type. (See updated PR description.)